dof_processor.py

```diff
@@ -0,0 +1,589 @@
+        date_str: Date in DD/MM/YYYY format
+        edition: Edition ('MAT' or 'VES')
+        
+    Returns:
+        ProcessStatus: SUCCESS if processing was successful, FAILED if failed, 
+        NO_FILES if no files to process, TIMEOUT if any document reached timeout
+    """
+    logging.info(f"Processing {date_str} - {edition}")
+    
+    # Search for DOC files
+    doc_files = find_doc_files_in_edition_folders(base_dir, date_str, edition)
+    
+    if not doc_files:
+        logging.info(f"No DOC files found for {date_str} - {edition}")
+        return ProcessStatus.NO_FILES  
+    
+    logging.info(f"Found {len(doc_files)} DOC files")
+    
+    # Phase 1: Convert DOC files to DOCX
+    docx_files = []
+    initial_problematic_count = len(get_problematic_files())
+    
+    for doc_file in doc_files:
+        docx_file = convert_doc_to_docx(doc_file, process_manager=_process_manager)
+        if docx_file and docx_file.exists():
+            docx_files.append(docx_file)
+            logging.info(f"Converted: {doc_file.name} -> {docx_file.name}")
+        else:
+            # Check if failure was due to timeout
+            current_problematic_count = len(get_problematic_files())
+            if current_problematic_count > initial_problematic_count:
+                logging.warning(f"TIMEOUT detected in {doc_file.name}. Skipping complete day: {date_str} - {edition}")
+                return ProcessStatus.TIMEOUT
+            logging.error(f"✗ Error converting: {doc_file.name}")
+    
+    if not docx_files:
+        logging.error(f"Could not convert DOC files for {date_str} - {edition}")
+        return ProcessStatus.FAILED
+    
+    # Phase 2: Merge DOCX documents
+    date_parts = date_str.split('/')
+    day, month, year = date_parts[0], date_parts[1], date_parts[2]
+    
+    unified_filename = f"{day}{month}{year}_{edition}.docx"
+    edition_dir = base_dir / year / month / f"{day}{month}{year}" / edition
+    unified_path = edition_dir / unified_filename
+    
+    logging.info(f"Merging {len(docx_files)} DOCX files...")
+    
+    if merge_docx_files(docx_files, unified_path):
+        logging.info(f"Merged document created: {unified_path}")
+        
+        # Phase 3: Automatic cleanup of residual files
+        logging.info("Starting cleanup of residual files...")
+        cleanup_temp_files(edition_dir)
+        
+        return ProcessStatus.SUCCESS
+    else:
+        logging.error(f"✗ Error creating merged document for {date_str} - {edition}")
+        return ProcessStatus.FAILED
+
+
+def main(
+    date: str = typer.Argument(..., help="Fecha (DD/MM/YYYY) o fecha de inicio para rango"),
+    end_date: Optional[str] = typer.Argument(None, help="Fecha de fin (DD/MM/YYYY) - opcional para rango de fechas"),
+    input_dir: str = typer.Option("./dof_word", help="Directorio de entrada donde buscar archivos DOC"),
+    editions: str = typer.Option("both", help="Ediciones a procesar: 'mat', 'ves', o 'both'"),
```

Comment:
In general, I think we shouldn't care about editions. We should just convert all files. I think not caring should simplify the code.

I see that the current folder structure is something like [year]/[month]/[date]/[edition]/files, and maybe instead we should have something resembling the PDFs folder structure: [year]/[month]/[date]-[MAT|VES]/files. That way, we could just process folders, we don't care about editions. But we already have all the doc files, so I maybe we should just deal with the folder structure we already have, but remove all the logic to proccess just one type or another and process ALL always.

What do you think?